okhttp/src/test/java/okhttp3/URLConnectionTest.java

```diff
@@ -2367,6 +2369,71 @@ private void testResponseRedirectedWithPost(int redirectCode, TransferKind trans
     testRedirect(false, "POST");
   }
 
+  class RevertInterceptor implements Interceptor {
+    @Override public Response intercept(Chain chain) throws IOException {
+      Response response = chain.proceed(chain.request());
+
+      return remapResponse(response);
+    }
+
+    private Response remapResponse(Response response) {
+      if (response.request().method().equals("POST") && (response.code() == HTTP_TEMP_REDIRECT || response.code() == HTTP_PERM_REDIRECT)) {
```

Comment:
oh this should probably be checking for not GET and not HEAD, otherwise PUT isn‘t consistent!